src/lib/rng/jitter_rng/jitter_rng.h

```diff
@@ -0,0 +1,41 @@
+/*
+* CPU Jitter Random Number Generator
+* (C) 2024 Planck Security S.A.
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#ifndef BOTAN_JITTER_RNG_H_
+#define BOTAN_JITTER_RNG_H_
+
+#include <botan/rng.h>
+
+namespace Botan {
+
+class BOTAN_DLL Jitter_RNG_Data;
+
+/*
+* RNG using libjitterentropy.
+*/
+class BOTAN_PUBLIC_API(3, 6) Jitter_RNG final : public RandomNumberGenerator {
+   public:
+      Jitter_RNG();
+      ~Jitter_RNG();
+
+      std::string name() const override;
+
+      bool is_seeded() const override;
+
+      bool accepts_input() const override;
+
+      void clear() override;
+
+   protected:
+      void fill_bytes_with_input(std::span<uint8_t> out, std::span<const uint8_t> in) override;
+
+   private:
+      Jitter_RNG_Data* m_jitter;
```

Comment:
There's a way to use `std::unique_ptr` here, like this:
`std::unique_ptr<Jitter_RNG_Data, void(Jitter_RNG_Data*)> m_jitter;`
Then just pass a function pointer (a lambda will do) where the `unique_ptr` is created in the `.cpp` file:
`Jitter_RNG::Jitter_RNG() : m_jitter{new Jitter_RNG_Data{}, [](Jitter_RNG_Data * p){ delete p; }} {}`

This approach is functionally equivalent to the c-like approach of defining a custom destruction function for a custom data structure; something like this:
```
// struct_a is not actually defined, but we can create a pointer to it
struct_a * my_a = create_struct_a();
// do something with the variable
destroy_struct_a(my_a);
```

Reply:
Yes, thank you, that helps the compiler. Pushed. I don't think I'm convinced that this simplifies things, plus there is technically still a naked `delete` in the code base, but it does avoid a destructor, and if it is preferred here then so be it, no problem.